src/components/LHNOptionsList/OptionRowLHNData.js

```diff
@@ -85,9 +92,19 @@ function OptionRowLHNData({
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [fullReport.reportID, receiptTransactions, reportActions]);
 
+    const hasViolations = canUseViolations && ReportUtils.doesTransactionThreadHaveViolations({report: fullReport, transactionViolations, parentReportAction});
+
     const optionItem = useMemo(() => {
         // Note: ideally we'd have this as a dependent selector in onyx!
-        const item = SidebarUtils.getOptionData(fullReport, reportActions, personalDetails, preferredLocale, policy, parentReportAction);
+        const item = SidebarUtils.getOptionData({
```

Comment:
Can we go back to individual params instead of a single object param? Or can you share why this change was made?

Reply:
There are a number of benefits to using an object for params over a long list of parameters.

For one, it means that the order of the parameters no longer matters, so if you need to add or remove a parameter in the future, it becomes a lot faster an easier, and when calling the function, you don't need to remember the order of the parameters.  

Also it means that every parameter is labeled with what it is. Which becomes important when there are long lists that include, for instance, boolean flags.

To understand`getNextAvailableDate(newDate(), true, true, false)` you need to know what those things mean, and it's also easy to transpose or forget one of the items.

But with:

```ts
getNextAvailableDate({
  startDate: newDate('2024-09-01'),
  useWeekends: true,
  allowDoubleBooking: false,
  cancelConflictingMeetings: false
  })
```

It's very clear what each of those values mean.

It also lets default values for boolean flags be omitted because you can define them as an optional boolean if they are rarely used. So the call can be reduced to:

```ts
getNextAvailableDate({
  startDate: newDate('2024-09-01'),
  useWeekends: true
})
```

In typescript you will get type checking on properties of objects so if something is not optional, then the compuiler will raise a type error if you omit a property from the object, the same as you would if you omitted a parameter.

On the other end, destructuring lets us access the params exactly as if they were written as separate params, and it maes things like default values a little clearer:

```ts
function getNextAvailableDate({
  startDate = newDate(), 
  useWeekends, 
  cancelConflictingMeetings, 
  allowDoubleBooking
  }:GetNextAvailableDateParams) {  
       // ...
  }
```

Reply:
That's fair, but the main reason I'm pushing back on this is because I'm trying to get all these PRs focused on the functionality and not add scope by refactoring. That said, I won't block on this, as I want to get this merged (which is another reason to limit the scope of the PR)

>For one, it means that the order of the parameters no longer matters

Doesn't TS have named params?

> For one, it means that the order of the parameters no longer matters

Do we encourage this kinda magic? I think I would prefer always having those values passed, or set with explicit destructuring, than magically have them set to false or to undefined which is falsy and have the code rely on that

Reply:
> I'm trying to get all these PRs focused on the functionality and not add scope by refactoring

This wasn't refactoring for the sake of refactoring, I had to do this to be able to work with these functions. Having to chase the flow of data through the various levels of callers was made far more annoying by dealing with long lists of params. The next person is going to have the same problem. 

> Doesn't TS have named params?

This is how we do that in typescript.

>  Do we encourage this kinda magic? 

There's no magic there -- this is bog standard JS/TS.

Reply:
I feel you, cause I use VIM. I assumed knowing the positions and names of parameters wasn't a problem for most people because they're editors make this easy. I actually downloaded visual studio code recently to make working in App a bit easier for myself

Reply:
I use webstorm which has parameter labels, but I still find it much easier to work with functions when the order of the params doesn't matter. 

For an extreme case say you have a function with 10 boolean params and you delete the seventh one. The labels are all wrong now, but there won't be any type errors or issues. And you need to go to each and count to the seventh param, and then make sure the seventh one is the right one to delete. But if you make a mistake counting at any point you've deleted the wrong thing and might not even notice until later.

When the params are in an object you can just follow the red squiggles and delete the one you know is the right one. 

Even passing in named variables can be wrong or worse misleading, because if the types are compatible, the compiler will accept it, but with an object using shorthand properties, you know that the variable with that name is being passed in the right place and will be consumed correctly by the called function.

With something like VSCode lables you need to do the work of matching each label to each variable name, and the compiler won't help you out.